### PR TITLE
[lld][WebAssembly] Add `--no-growable-memory`

### LINKED ISSUE #81932: [wasm-ld] How to specify a non-growable memory without `--initial-memory` and `--max-memory`
Until recently the way to specify a non-growable memory was to specify both `--initial-memory` and `--max-memory` and set them to the same value.   This is what emscripten currently does by default unless you opt into `-sALLOW_MEMORY_GROWTH`.

However, we recently added a new `--initial-heap` flag which allows the user to set and initial heap size and have the linker then derive the initial memory size based on the sum of stack size + heap size + static data size.  See #75594.

When attempting to start using this flag in emscripten we discovered that there is no way to use the `--initial-heap` flag and still get a non-growable memory.  See https://github.com/emscripten-core/emscripten/pull/21071/files#r1491731562.

It seems that we need some kind of flag like the existing `--growable-table` flag.  It would be nice to add `--growable-memory` except that would mean changing the default to non-growable, which is perhaps too much of a breaking change.  @sunfishcode WDYT?

Instead perhaps we can add `--no-growable-memory`?  

Perhaps the existing `--growable-table` flag could also take a `no-` prefix so that they were symmetrical?

### PR DESCRIPTION
**Pull request by SingleAccretion**
We recently added `--initial-heap` - an option that allows one to up the initial memory size without the burden of having to know exactly how much is needed.

However, in the process of implementing support for this in Emscripten (https://github.com/emscripten-core/emscripten/pull/21071), we have realized that `--initial-heap` cannot support the use-case of non-growable memories by itself, since with it we don't know what to set `--max-memory` to.

We have thus agreed to move the above work forward by introducing another option to the linker (see https://github.com/emscripten-core/emscripten/pull/21071#discussion_r1491755616), one that would allow users to explicitly specify they want a non-growable memory.

This change does this by introducing `--no-growable-memory`: an option that is mutally exclusive with `--max-memory` (for simplicity - we can also decide that it should override or be overridable by `--max-memory`. In Emscripten a similar mix of options results in `--no-growable-memory` taking precedence). The option specifies that the maximum memory size should be set to the initial memory size, effectively disallowing memory growth.

Closes #81932.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@sbc100

**Comment by sbc100** [comment]
How about adding `--[no-]-growable-memory` instead?   It seems like a better match the existing `--growable-table` option and seem more obviously self-explantiory.

File: lld/wasm/Writer.cpp [hunk 1]
```diff
@@ -473,6 +473,12 @@ void Writer::layoutMemory() {
     WasmSym::heapEnd->setVA(memoryPtr);
   }
 
+  if (config->maxMemory != 0 && config->noGrowableMemory) {
+    // Erroring out here is simpler than defining precedence rules.
+    error("--max-memory and --no-growable-memory are mutually exclusive");
+  }
```
Comment (sbc100) [inline lld/wasm/Writer.cpp:479]: Can you do this error checking in `readConfigs`?

There are a bunch of other error message there of the form `xxx is incompatible with yyy`.  Can we also use that terminology for consistency?

**Review by sbc100 — COMMENTED** [review]
lgtm with one more comment

**Review by sbc100 — APPROVED** [review]

**Pull request merged**